pkg/apis/config/defaults.go

```diff
@@ -68,65 +68,185 @@ func NewDefaultsConfigFromConfigMap(config *corev1.ConfigMap) (*Defaults, error)
-type ClassAndBrokerConfig struct {
-	BrokerClass   string `json:"brokerClass,omitempty"`
-	*BrokerConfig `json:",inline"`
+// DefaultConfig struct contains the default configuration for the cluster and namespace.
+type DefaultConfig struct {
+	// DefaultBrokerClass and DefaultBrokerClassConfig are the default broker class for the whole cluster/namespace.
+	// Users have to specify both of them
+	DefaultBrokerClass string `json:"brokerClass,omitempty"`
+	*BrokerConfig      `json:",inline"`
+
+	// Optional: BrokerClasses are the default broker classes' config. The key is the broker class name, and the value is the config for that broker class.
+	BrokerClasses map[string]*BrokerConfig `json:"brokerClasses,omitempty"`
 
 	DisallowDifferentNamespaceConfig *bool `json:"disallowDifferentNamespaceConfig,omitempty"`
 }
 
-// BrokerConfig contains configuration for a given namespace for broker. Allows
+// BrokerConfig contains configuration for a given broker. Allows
 // configuring the reference to the
 // config it should use and it's delivery.
 type BrokerConfig struct {
 	*duckv1.KReference `json:",inline"`
 	Delivery           *eventingduckv1.DeliverySpec `json:"delivery,omitempty"`
 }
 
-// GetBrokerConfig returns a namespace specific Broker Configuration, and if
+// GetBrokerConfig returns a namespace specific Broker Config, and if
 // that doesn't exist, return a Cluster Default and if that doesn't exist
-// return an error.
-func (d *Defaults) GetBrokerConfig(ns string) (*BrokerConfig, error) {
+func (d *Defaults) GetBrokerConfig(ns string, brokerClassName *string) (*BrokerConfig, error) {
 	if d == nil {
 		return nil, errors.New("Defaults are nil")
 	}
-	value, present := d.NamespaceDefaultsConfig[ns]
-	if present && value.BrokerConfig != nil {
-		return value.BrokerConfig, nil
+
+	// Early return if brokerClassName is provided and valid
+	if brokerClassName != nil && *brokerClassName != "" {
+		return d.getBrokerConfigByClassName(ns, *brokerClassName)
 	}
-	if d.ClusterDefault != nil && d.ClusterDefault.BrokerConfig != nil {
-		return d.ClusterDefault.BrokerConfig, nil
+
+	// Handling empty brokerClassName
+	return d.getBrokerConfigForEmptyClassName(ns)
+}
+
+// getBrokerConfigByClassName returns the BrokerConfig for the given brokerClassName.
+// It first checks the namespace specific configuration, then the cluster default configuration.
+func (d *Defaults) getBrokerConfigByClassName(ns string, brokerClassName string) (*BrokerConfig, error) {
+	// Check namespace specific configuration
+	if nsConfig, ok := d.NamespaceDefaultsConfig[ns]; ok && nsConfig != nil {
+		// check if the brokerClassName is the default broker class for this namespace
+		if nsConfig.DefaultBrokerClass == brokerClassName {
+			if nsConfig.BrokerConfig == nil {
+				// as no default broker class config is set for this namespace, should get the cluster default config
+				return d.getClusterDefaultBrokerConfig(brokerClassName)
+			}
```

Comment:
what happens if:
1. the user specifies the default broker class
2. the user does not specify the default broker config
3. the user provides a config for the broker class in the map of extra broker classes and configs

Would it make sense to check if `nsConfig.BrokerClasses[brokerClassName]` has a value?